synapse/handlers/federation.py

```diff
@@ -583,7 +583,12 @@ async def do_invite_join(
                 # Mark the room as having partial state.
                 # The background process is responsible for unmarking this flag,
                 # even if the join fails.
-                await self.store.store_partial_state_room(room_id, ret.servers_in_room)
+                await self.store.store_partial_state_room(
+                    room_id,
+                    ret.servers_in_room,
+                    ret.event.event_id,
```

Comment:
Can we insert null here, and then update it with the join event ID in the persist event DB transaction? That way we might be able to keep the foreign key constraint?

Reply:
It'd mean that we have to tell the persist event logic "you're persisting a partial join response" somehow. I can look at doing this but I'm not sure if it's worth the extra complexity in the persistence path. WDYT?